distributed_nhc/export_nhc_result_to_kusto.py

```diff
@@ -60,13 +63,177 @@ def ingest_debug_log(debug_file, creds, ingest_url, database, debug_table_name):
+
+        # Split the string by lines and create key-value pairs
+        for line in data_string.strip().split("\n"):
+            key, value = line.split(":")
+            if key.startswith("ib_write_lb_mlx5_ib"):
+                result["IB_WRITE_GDR"][key] = str(value.strip())
+            elif key.startswith("H2D"):
+                result["GPU_BW_HTD"][key] = str(value.strip())
+            elif key.startswith("D2H"):
+                result["GPU_BW_DTH"][key] = str(value.strip())
+            elif key.startswith("P2P"):
+                result["GPU_BW_P2P"][key] = str(value.strip())
+            elif key.startswith("nccl_all_red"):
+                result["NCCL_ALL_REDUCE"] = str(value.strip())
+            elif key.startswith("nccl_all_red_lb"):
+                result["NCCL_ALL_REDUCE_LOOP_BACK"] = str(value.strip())
+    
+    else : 
+        result = {"Error": "Unable to determine processor type"}
+
+    return result
+
+def ingest_results(results_file, creds, ingest_url, database, results_table_name, nhc_run_uuid="None"):
+    ts = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+
+    job_name = results_file.replace("\\", "/").split(".")[0].split("/")[-1] # account for \ or / in path
+    uuid = job_name if nhc_run_uuid == "None" else f"{nhc_run_uuid}-{job_name}"
+    if uuid == "health":
+        uuid = ""
+    else :
+        uuid = "-" + uuid # add the dash here instead of below; this way if 'uuid' is empty, we don't have a trailing dash
+    full_uuid = f"nhc-{ts}{uuid}"
+
+    vmSize_bash_cmd = "echo $( curl -H Metadata:true --max-time 10 -s \"http://169.254.169.254/metadata/instance/compute/vmSize?api-version=2021-01-01&format=text\") | tr '[:upper:]' '[:lower:]' "
+    vmSize = run_command(vmSize_bash_cmd)
+
+    vmId_bash_cmd = "curl  -H Metadata:true --max-time 10 -s \"http://169.254.169.254/metadata/instance/compute/vmId?api-version=2021-02-01&format=text\""
+    vmId = run_command(vmId_bash_cmd)
+
+    vmName_bash_cmd = "hostname"
+    vmName = run_command(vmName_bash_cmd)
+
+    physhost = run_command("echo $(hostname) \"$(/opt/azurehpc/tools/kvp_client | grep Fully)\" | cut -d ':' -f 3 | cut -d ' ' -f 2 | sed 's/\"//g'")
+    if not physhost:
+        physhost = "not Mapped"
+
+    with open(results_file, 'r') as f:
+        full_results = f.read()
+        jsonResultDict = get_nhc_json_formatted_result(results_file)
+        jsonResult = json.dumps(jsonResultDict)
+
+        record = {
+            'vmSize': vmSize,
+            'vmId': vmId,
+            'vmHostname': vmName,
+            'physHostname': physhost,
+            'workflowType': "main",
+            'time': ts,
+            'pass': False, # keep as default false
+            'error': '',
+            'logOutput': full_results, # the entire file
+            'jsonResult': jsonResult,
+            'uuid': full_uuid
+        }
+
+        if "Node Health Check completed successfully " in full_results:
```

Comment:
Do you need the whitespace after "successfully"?

Reply:
Won't cause issues as there's some things that show up afterwards, but don't need it; removed!